local_sales/app/controllers/towns_controller.rb

```diff
@@ -0,0 +1,52 @@
+class TownsController < ApplicationController
+  before_action :set_town, only: %i[edit update]
+
+  # GET /states
+  def index
+    @town = Town.all
+    @townships = Township.all    
```

Comment:
I see that you are not using this variable, Is it needed?

Reply:
i thought that i needed to call the name of the township, i'll fix in a new issue for bugs/unnecessary things